File: api/src/main/java/traductorfrases/api/aplicacion/servicios/IdiomaServicio.java
package traductorfrases.api.aplicacion.servicios;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import traductorfrases.api.dominio.entidades.Idioma;
import traductorfrases.api.infraestructura.repositorios.IIdiomaRepositorio;

@Service
public class IdiomaServicio {

    @Autowired
    IIdiomaRepositorio repositorio;

    public List<Idioma> obtenerTodos() {
        return repositorio.findAll(Sort.by("nombre"));
    }

    public Idioma obtener(int id) {
        return repositorio.findById(id).get();
    }

    public Idioma agregar(Idioma idioma) {
        idioma.setId(0);
        return repositorio.save(idioma);
    }

    public Idioma modificar(Idioma idioma) {
        return repositorio.findById(idioma.getId()).isPresent() ? repositorio.save(idioma) : null;
    }

    public boolean eliminar(int id) {
        if (repositorio.findById(id).isPresent()) {
            try {
                repositorio.deleteById(id);
                return true;
            } catch (Exception e) {
                return false;
            }
        }
        return false;
    }

}

File: TraduccionFrases/js/servicios/idioma.servicio.js
app.service("IdiomaServicio", function ($http) {
    const URL = "http://localhost:8080/api/idiomas/";

    this.listar = () => {
        return $http.get(URL).then(respuesta => respuesta.data);
    };

    this.obtener = (id) => {
        return $http.get(`${URL}${id}`).then(respuesta => respuesta.data);
    };

    this.agregar = (idioma) => {
        return $http.post(URL, idioma).then(respuesta => respuesta.data);
    };

    this.modificar = (idioma) => {
        return $http.put(URL, idioma).then(respuesta => respuesta.data);
    };

    this.eliminar = (id) => {
        return $http.delete(`${URL}${id}`).then(respuesta => respuesta.data);
    };
});